### PR TITLE
test done

### PR DESCRIPTION
**Pull request by BondarenkoMaks**

### CODE REVIEW

File: src/components/MainPage/index.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+class UserTable extends Component {
+	static propTypes = {
+		users: PropTypes.array.isRequired,		
+		toggleReset: PropTypes.func
+	}
+
+	constructor(props) {
+		super(props);
+		debugger;
```
Comment (KatSick) [inline src/components/MainPage/index.js]: I think, this code should not be here

File: src/components/MainPage/index.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+		const { loaded, loading, dispatch } = this.props;
+
+		if (!loaded && !loading) dispatch(loadAllUsers());
+	}
+
+	
+	render() {
+		const { users } = this.props;
+		debugger;
```
Comment (KatSick) [inline src/components/MainPage/index.js]: This debugger is irrelevant

File: src/components/MainPage/index.js [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+					</div>
+				</div>
+			</div>
+
+		);
+	}
+}
+
+const mapStateToProps = (state, props) => {
+	debugger;
```
Comment (KatSick) [inline src/components/MainPage/index.js]: This debugger is irrelevant

File: src/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+
+import { LOAD_ALL_USERS, SUCCESS, START, FAIL } from './../constants';
+
+export const loadAllUsers = () => {
+	return (dispatch, getState) => {
```
Comment (KatSick) [inline src/actions/index.js]: getState ia not used here

File: src/changeDataStructure.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        }else{
+           return acc.set(item.uid, new DataRecord(item));
+        }
+    }
+       
+    , new OrderedMap({}))
+}
+
+export function mapToArr(obj) {
+    //console.log(obj);
```
Comment (KatSick) [inline src/changeDataStructure.js]: This comment is unnecessary

File: src/changeDataStructure.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    
+   return obj.valueSeq().toArray();
+    //return obj;
```
Comment (KatSick) [inline src/changeDataStructure.js]: This comment is unnecessary

File: src/components/App/index.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+					
+					<div  id="logo" className="center-block"><img src="images/strikersoft.svg" /> </div>
+					<MainPage />
+				</div>
+			</div>
+		);
+	}
+}
+
+const mapStateToProps = (state) => {
```
Comment (KatSick) [inline src/components/App/index.js]: Do we need empty mapStateToProps?

File: src/components/App/index.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	return {
+
+	}
+};
+
+const mapDispatchtoProps = (dispatch) => {
```
Comment (KatSick) [inline src/components/App/index.js]: Do we need empty mapDispatchToProps?

File: src/components/MainPage/index.js [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+	}
+	handleReset(ev) {
+		this.setState({
+			sortByName: false,
+			sortByAge: false,
+			currentUser: null
+
+		});
+		let { toggleReset } = this.props;
+		toggleReset(); // reset in decorator
```
Comment (KatSick) [inline src/components/MainPage/index.js]: Why do we need HOC toggleSort? Is it re-used anywhere else?

File: src/components/MainPage/style.scss [hunk 10]
```diff
@@ -0,0 +1,56 @@
+.box{
+    // background-color: white;
+    
+    margin: 10px;
+    margin-left: 0;
+    margin-right: 0;
+    border-radius: 8px;
+    // padding: 21px;
```
Comment (KatSick) [inline src/components/MainPage/style.scss]: Do we need to keep this comment? Will it help our codebase?

File: src/changeDataStructure.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import {OrderedMap, Map} from 'immutable';
+
+export function arrToMap(arr, DataRecord = Map) {
```
Comment (KatSick) [inline src/changeDataStructure.js:3]: What is the benefit of this normalization of data, and why this function re-used from here ? https://github.com/bars/react-learnjs/blob/master/js_ru_08_06_17-master/src/helpers.js

**Review by KatSick — CHANGES_REQUESTED** [review]
Hi, please, resolve my comments before we continue

File: src/components/MainPage/index.js (hunk 9 above)
Comment (BondarenkoMaks) [reply src/components/MainPage/index.js]: That needs to change a parent state and get previous list users

Comment (KatSick) [reply src/components/MainPage/index.js]: @BondarenkoMaks thanks for your explanation! I understand what `toggleReset` is doing. However, my question concerns using HOC pattern here.

File: src/changeDataStructure.js (hunk 11 above)
Comment (BondarenkoMaks) [reply src/changeDataStructure.js:3]: i studied this course and used it because a record is more convenient that list

Comment (KatSick) [reply src/changeDataStructure.js:3]: Thanks for your explanation

**Pull request closed without merging**